modules/core/src/main/forum.scala

```diff
@@ -3,16 +3,38 @@ package forum
 
 import lila.core.id.{ ForumCategId, ForumPostId, ForumTopicId, TeamId }
 import lila.core.userId.*
+import lila.core.bus
 
 import reactivemongo.api.bson.Macros.Annotations.Key
 
-case class CreatePost(post: ForumPostMini)
+sealed class BusForumPost
+object BusForumPost:
+  given busForumPostChannel: bus.WithChannel[BusForumPost] with
+    def channel = "forumPost"
+
+case class CreatePost(post: ForumPostMini) extends BusForumPost
+object CreatePost:
+  given createPostChannel: bus.WithChannel[CreatePost] with
+    def channel = "forumPost"
 case class RemovePost(id: ForumPostId, by: Option[UserId], text: String, asAdmin: Boolean)(using val me: MyId)
-case class RemovePosts(ids: List[ForumPostId])
+    extends BusForumPost
+object RemovePost:
+  given removePostChannel: bus.WithChannel[RemovePost] with
+    def channel = "forumPost"
+case class RemovePosts(ids: List[ForumPostId]) extends BusForumPost
+object RemovePosts:
+  given removePostsChannel: bus.WithChannel[RemovePosts] with
+    def channel = "forumPost"
 // case class PostCloseToggle(categ: ForumCategId, topicSlug: String, closed: Boolean)(using val me: user.MyId)
 // erasing = blankng, still in db but with empty text
-case class ErasePost(id: ForumPostId)
-case class ErasePosts(ids: List[ForumPostId])
+case class ErasePost(id: ForumPostId) extends BusForumPost
+object ErasePost:
+  given erasePostChannel: bus.WithChannel[ErasePost] with
+    def channel = "forumPost"
+case class ErasePosts(ids: List[ForumPostId]) extends BusForumPost
+object ErasePosts:
+  given erasePostsChannel: bus.WithChannel[ErasePosts] with
+    def channel = "forumPost"
```

Comment:
Massive boilterplate to add given for each type, can probably be refactored or at least generated with a macro

Reply:
I think We don't need to add an extra `given` for every `case class`, because We already have the `given` in `sealed class BusForumPost`.

And then We can just use `enum` here to make it less verbose.

Reply:
so just to be clear, turn `BusForumPost` into an enum with each case class being a member of it, right?

The `given` for each class was necessary for the compiler to find when `bus.pub`-ing.

Reply:
> so just to be clear, turn `BusForumPost` into an enum with each case class being a member of it, right?
👍 

> The `given` for each class was necessary for the compiler to find when `bus.pub`-ing.

No, I don't think it's necessary because We already have that in `BusForumPost`.
maybe We need to change a little bit though:

```scala
object BusForumPost:
  given [T <: BusForumPost]: bus.WithChannel[T] with
    def channel = "forumPost"
```